rocketpy/Rocket.py

```diff
@@ -537,56 +528,133 @@ def addFins(
+
+        # Type verification
+        if type == "trapezoid":
+
+            # Retrieve parameters for calculations
+            Ct = finParameters["tipChord"]
+            Yr = Cr + Ct
+
+            Af = Yr * s / 2  # fin area
+            gamac = np.arctan((Cr - Ct) / (2 * s))  # mid chord angle
+            Yma = (
+                (s / 3) * (Cr + 2 * Ct) / Yr
+            )  # span wise position of fin's mean aerodynamic chord
+            Lf = np.sqrt((Cr / 2 - Ct / 2) ** 2 + s**2)
+
+            rollGeometricalConstant = (
+                (Cr + 3 * Ct) * s**3
+                + 4 * (Cr + 2 * Ct) * radius * s**2
+                + 6 * (Cr + Ct) * s * radius**2
+            ) / 12
+
+            # Calculate cp position relative to cm
+            if distanceToCM < 0:
+                cpz = distanceToCM - (
+                    ((Cr - Ct) / 3) * ((Cr + 2 * Ct) / (Cr + Ct))
+                    + (1 / 6) * (Cr + Ct - Cr * Ct / (Cr + Ct))
+                )
+            else:
+                cpz = distanceToCM + (
+                    ((Cr - Ct) / 3) * ((Cr + 2 * Ct) / (Cr + Ct))
+                    + (1 / 6) * (Cr + Ct - Cr * Ct / (Cr + Ct))
+                )
+
+        elif type == "elliptical":
+
+            # Retrieve parameters for calculations
+            Af = (np.pi * Cr / 2 * s) / 2
+            gamac = 0
+            Yma = s / (3 * np.pi) * np.sqrt(9 * np.pi**2 - 64)
+            rollGeometricalConstant = (
+                Cr
+                * s
+                * (3 * np.pi * s**2 + 32 * radius * s + 12 * np.pi * radius**2)
+                / 48
+            )
+
+            # Calculate cp position relative to cm
+            if distanceToCM < 0:
+                cpz = distanceToCM - (0.288 * Cr)
+            else:
+                cpz = distanceToCM + (0.288 * Cr)
+
+        else:
+            raise Exception("Invalid fin shape")
+
+        # Retrieves abstract parameters
         Aref = np.pi * radius**2
-        AR = 2 * s**2 / Af  # Barrowman's convention for fin's aspect ratio
-        cantAngleRad = np.radians(cantAngle)
-        trapezoidalConstant = (
-            (Cr + 3 * Ct) * s**3
-            + 4 * (Cr + 2 * Ct) * radius * s**2
-            + 6 * (Cr + Ct) * s * radius**2
-        ) / 12
+        AR = 2 * s**2 / Af
+        rollParameters = [0, 0, 0]
 
         # Fin–body interference correction parameters
-        τ = (s + radius) / radius
-        λ = Ct / Cr
-        liftInterferenceFactor = 1 + 1 / τ
+        tau = (s + radius) / radius
+        liftInterferenceFactor = 1 + 1 / tau
+        if type == "trapezoid":
+            λ = Ct / Cr
+            rollDampingInterferenceFactor = 1 + (
+                ((tau - λ) / (tau)) - ((1 - λ) / (tau - 1)) * np.log(tau)
+            ) / (
+                ((tau + 1) * (tau - λ)) / (2)
+                - ((1 - λ) * (tau**3 - 1)) / (3 * (tau - 1))
+            )
+        else:  # type == "elliptical"
+            rollDampingInterferenceFactor = 1 + (
```

Comment:
this equation is different from (16) on your document. Can you give just a double check on every term?